build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,239 @@
+	}
+
+	return {
+		result: false,
+		message: `Prerelease dependencies found. These release groups and packages must be released and integrated before a release can be done:\n\n${[
+			...releaseGroups.keys(),
+		].join("\n")}\n${[...packages.keys()].join("\n")}`,
+		fixCommand: "pnpm flub bump deps <release group or package>",
+	};
+};
+
+const checkPolicy: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	// policy-check is scoped to the path that it's run in. Since we have multiple folders at the root that represent
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	const result = await execa.command(`npm run policy-check`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	if (result.exitCode !== 0) {
+		return {
+			result: false,
+			message: "Policy check failed.",
+			fixCommand: "pnpm run policy-check:fix",
+		};
+	}
+
+	return { result: true };
+};
+
+const checkAsserts: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	// policy-check is scoped to the path that it's run in. Since we have multiple folders at the root that represent
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	await execa.command(`npm run policy-check:asserts`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	// check for policy check violation
+	const afterPolicyCheckStatus = await context.gitRepo.getStatus();
+	if (afterPolicyCheckStatus !== "") {
+		const repo = new Repository({ baseDir: context.repo.resolvedRoot });
+		await repo.gitClient.reset(ResetMode.HARD);
+		return {
+			result: false,
+			message: "Found some untagged asserts. These should be tagged before release.",
+			fixCommand: "pnpm run policy-check:asserts",
+		};
+	}
+
+	return { result: true };
+};
+
+const allChecks = new Map<string, CheckFunction>([
+	["Branch has no local changes", checkNoLocalChanges],
+	["Has a microsoft/FluidFramework remote", checkHasRemote],
+	["The local branch is up to date with the remote", checkBranchUpToDate],
+	["Dependencies are installed locally", checkDepsInstalled],
+	["Has no pre-release Fluid dependencies", checkHasNoPrereleaseDependencies],
+	["No repo policy violations", checkPolicy],
+	["No untagged asserts", checkAsserts],
+]);
+
+const packageOrReleaseGroupArg = clonedeep(baseArg);
+packageOrReleaseGroupArg.description =
+	"The name of a package or a release group. Defaults to the client release group if not specified.";
+packageOrReleaseGroupArg.default = "client";
+
+export class ReleasePrepareCommand extends BaseCommand<typeof ReleasePrepareCommand> {
+	static summary = `Runs checks on a local branch to verify it is ready to serve as the base for a release branch.`;
+
+	static description = `Runs the following checks:\n\n- ${[...allChecks.keys()].join("\n- ")}`;
+
+	static aliases: string[] = ["release:prep"];
+
+	static args = {
```

Comment:
mutable statics are creepy (changes can last forever and impact unrelated use of a module). Can you make these readonly? Also the allChecks could be made a readonly map,

Reply:
I updated these to be readonly, and changed the shared CLI argument into a factory function so its easier to customize
the argument metadata. I didn't make the readonly change across all commands - only those that were updated for other
reasons in this PR.

Reply:
#17680 updates all commands to use `static readonly`.

Reply:
This pattern has since been corrected across the build-tools packages.